### DISCUSSION TITLE
Release 1.2.0 Beta 4

### DISCUSSION THREAD

**Discussion by apohl79** [apohl79/audiogridder#652] [General]
## GENERAL
- Improved I/O support
  - Multiple instrument outputs with up to 64 channels (resolves #597)
    - Individual instrument outputs can be enabled/disabled and can be routed to track inputs
  - Each FX or instrument input/output channel can be enabled/disabled (resolves #130)
  - Multi-channel sidechains (>2 channels)
  - More flexible I/O channel mapping (resolves #608)
- Improved mouse capture/replay (resolves #500, resolves #501)
- Added options to enable/disable CoreDumps (resolves #544)

## PLUGIN
- Minor UI fixes

## SERVER
- Minor stability fixes

<hr /><em>This discussion was created from the release <a href='https://github.com/apohl79/audiogridder/releases/tag/release_1_2_0_beta4'>Release 1.2.0 Beta 4</a>.</em>

**Comment by emack999** [comment]
Thanks for the latest beta.
Windows version, AG Midi doesn't find the server in beta4.
Tried with VST and VST3, neither would find the server.
It says its connected in the SERVER section of the context menu...
https://imgur.com/a/i9MIRls

But no connection is shown in the plugin side, no plugins are listed, and the I/O in the monitor window shows 0/0.
https://imgur.com/a/ldCSjut

The AG and AG Instrument plugins find the server and list the plugins just fine with both VST and VST3, with the correct I/O also.  It's just the AG Midi version of the plugin that is affected.  

Tested with Gig Performer 4 and Cubase 9.5

**Reply by apohl79** [reply]
thx, will have a look

**Comment by Pete3030** [comment]
One question, is it recommended to update to the newest Beta (from V1.1 in my case) when I already use the old version of AG in a current project, means, does AG remember the plugin chains and (even more important) the plugin settings, or do I have to set it up again completely from the scratch after the update?

**Reply by apohl79** [reply]
AG stays backwards compatible. You can also downgrade without destroying your session. But you can do a backup of your session, to be on the safe side.

**Comment by Pete3030** [comment]
OK, I will do that, thanks a lot for your feedback!

**Comment by BubsonGH** [comment]
Just a small bug report for 1.2 beta 4 apposed to using 1.1

When I was using 1.1 yesterday with the plugin Flatline from submission audio it worked fine, after updating to beta 4.. when I click to open the plugin it opens and closes multiple times on its own and once it has opened, the visual graph feedback doesn't show unless I close it and open it again 1 time. but when the graph does show it is also messed up and not showing correctly.

**Comment by lesbrunn** [comment]
> ## GENERAL
> * Improved I/O support
>   
>   * Multiple instrument outputs with up to 64 channels (resolves [Improve Multi-Output Support #597](https://github.com/apohl79/audiogridder/issues/597))
>     
>     * Individual instrument outputs can be enabled/disabled and can be routed to track inputs
>   * Each FX or instrument input/output channel can be enabled/disabled (resolves [Request: mono switch #130](https://github.com/apohl79/audiogridder/issues/130))
>   * Multi-channel sidechains (>2 channels)
>   * More flexible I/O channel mapping (resolves ["failed to find working I/O configuration" error message #608](https://github.com/apohl79/audiogridder/issues/608))
> * Improved mouse capture/replay (resolves [KONTAKT Library Browsing problem in AG Plugin Side #500](https://github.com/apohl79/audiogridder/discussions/500), resolves [KONTAKT Library Loading Problem....!! #501](https://github.com/apohl79/audiogridder/issues/501))
> * Added options to enable/disable CoreDumps (resolves [Add option to disable core dumps #544](https://github.com/apohl79/audiogridder/issues/544))
> 
> ## PLUGIN
> * Minor UI fixes
> 
> ## SERVER
> * Minor stability fixes
> 
> _This discussion was created from the release [Release 1.2.0 Beta 4](https://github.com/apohl79/audiogridder/releases/tag/release_1_2_0_beta4)._

Hi Andreas,
Well done! I didn't see this thread before, so I posted this in the 'Multiple Outputs' thread. This is my comment there:

 ''After not using AG for some time, I tested the new beta yesterday. I wanted to check out the multiple outputs, so I loaded SSD5, which I 
 use quite often. I successfully used up to 8 stereo outputs with 2 blocks at a buffer of 64 in Pro Tools. Latency wasn't too bad, but I 
 had several stalls and crashes, and I had to restart AG several times. The main issue, though, is that with only SSD5 in PT at the above- 
 mentioned settings, PT's CPU meter was very unstable and sometimes shot up to 48%. Disabling AG brought the meter down to 1%. 
 Loading SSD5 with the same 8 outputs in PT without AG gave me a reading of only 4% at the highest peak. As solid as the AG concept is, 
 the purpose of reducing the DAW's CPU load isn't yet happening properly. There's still a lot of work to be done for AG to lift the load off 
 PT. To be fair, though, I haven't yet tested AG with other VI's. I intend to do so when I have some time, and I'll post my findings.''

**Reply by lesbrunn** [reply]
Does anyone else feel that CPU usage in their DAW increases when AG is instantiated? Or is it something in my system? I need to do more testing, but it seems to me that the issue is more with Vi's than with other types of plugins. I may be wrong on this, though.
